server/src/main/java/org/opensearch/index/translog/RemoteFsTimestampAwareTranslog.java

```diff
@@ -241,14 +251,8 @@ public void onFailure(Exception e) {
     protected Set<Long> getGenerationsToBeDeleted(
         List<String> metadataFilesNotToBeDeleted,
         List<String> metadataFilesToBeDeleted,
-        boolean indexDeleted
+        long minRemoteGenReferenced
     ) throws IOException {
-        long maxGenerationToBeDeleted = Long.MAX_VALUE;
-
-        if (indexDeleted == false) {
-            maxGenerationToBeDeleted = minRemoteGenReferenced - 1 - indexSettings().getRemoteTranslogExtraKeep();
```

Comment:
are we not keeping extra generations anymore?

Reply:
As I think more, we still need this setting. Thanks for calling out. I will make the change.